install.sh

```diff
@@ -0,0 +1,207 @@
+  fi
+}
+
+OS="$(determine_os)"
+
+function determine_arch() {
+  # Map platform architecture to released binary architecture.
+  read_arch="$(uname -m)"
+  case "$read_arch" in
+    x86_64*)   echo "x86_64" ;;
+    amd64*)    echo "x86_64" ;;
+    arm64*)    echo "aarch64" ;;
+    aarch64*)  echo "aarch64" ;;
+    *)        die "unknown arch: ${read_arch}" ;;
+  esac
+}
+
+ensure_cmd basename
+ensure_cmd $([[ "${OS}" == "windows" ]] && echo "pwsh" || echo "curl")
+function fetch() {
+  local url="$1"
+  local dest_dir="$2"
+
+  local dest
+  dest="${dest_dir}/$(basename "${url}")"
+
+  if [[ "${OS}" == "windows" ]]; then
+    pwsh -c "Invoke-WebRequest -OutFile ${dest} -Uri ${url}"
+  else
+    curl --proto '=https' --tlsv1.2 -SfL --progress-bar -o "${dest}" "${url}"
+  fi
+}
+
+ensure_cmd $([[ "${OS}" == "macos" ]] && echo "shasum" || echo "sha256sum")
+function sha256() {
+  if [[ "${OS}" == "macos" ]]; then
+    shasum --algorithm 256 "$@"
+  else
+    sha256sum "$@"
+  fi
+}
+
+ensure_cmd dirname
+ensure_cmd mktemp
+ensure_cmd install
+function install_from_url() {
+  local url="$1"
+  local dest="$2"
+
+  local workdir
+  workdir="$(mktemp -d)"
+  gc "${workdir}"
+
+  fetch "${url}.sha256" "${workdir}"
+  fetch "${url}" "${workdir}" && green "Download completed successfully"
+  (
+    cd "${workdir}"
+    sha256 -c --status ./*.sha256 &&
+      green "Download matched it's expected sha256 fingerprint, proceeding" ||
+        die "Download from ${url} did not match the fingerprint at ${url}.sha256"
+  )
+  rm "${workdir}/"*.sha256
+
+  if [[ "${OS}" == "macos" ]]; then
+    mkdir -p "$(dirname "${dest}")"
+    install -m 755 "${workdir}/"* "${dest}"
+  else
+    install -D -m 755 "${workdir}/"* "${dest}"
+  fi
+
+  green "Installed ${url} to ${dest}"
+}
+
+ensure_cmd cat
+function usage() {
+  cat << __EOF__
+Usage: $0
+
+Installs the \`science\` binary.
+
+-h | --help: Print this help message.
+
+-d | --bin-dir:
+  The directory to install the science binary in, "~/.local/bin" by default.
+
+-b | --base-name:
+  The name to use for the science binary, "science" by default.
+
+-V | --version:
+  The version of the science binary to install, the latest version by default.
+  The available versions can be seen at:
+    https://github.com/a-scie/lift/releases
+
+__EOF__
+}
+
+INSTALL_PREFIX="${HOME}/.local/bin"
+INSTALL_FILE="science"
+VERSION="latest/download"
+
+# Parse arguments.
+while (($# > 0)); do
+  case "$1" in
+    --help | -h)
+      usage
+      exit 0
+      ;;
+    --bin-dir | -d)
+      INSTALL_PREFIX="$2"
+      shift
+      ;;
+    --base-name | -b)
+      INSTALL_FILE="$2"
+      shift
+      ;;
+    --version | -V)
+      VERSION="download/v${2}"
+      shift
+      ;;
+    *)
+      usage
+      die "Unexpected argument: ${1}\n"
+      ;;
+  esac
+  shift
+done
+
+ARCH="$(determine_arch)"
+INSTALL_DEST="${INSTALL_PREFIX}/${INSTALL_FILE}"
+DL_URL="https://github.com/a-scie/lift/releases/${VERSION}/science-fat-${OS}-${ARCH}"
```

Comment:
Aha - Ok. Yeah, just a bad URL for the Windows case. Should be good to go after that fix.

Reply:
You took my meaning wrong. You're missing adding an .exe on the end of this URL for Windows. There _are_ Windows releases for both x86-64 and aarch64, search for ".exe" here: https://github.com/a-scie/lift/releases/tag/v0.7.0

You'll need to bump you 0.6.1 to 0.7.0 though so that the tests work on Windows arm64.

Reply:
aha, I was looking at https://github.com/a-scie/lift/releases and the lack of x86_64 (need to click "show all") was a red herring. thanks!